src/components/Api.js

```diff
@@ -0,0 +1,108 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  getUserInfoAndCards() {
+    return Promise.all([this.getInitialCards(), this.getUser()]);
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  //-----CARD ROUTE: Get all cards (GET)----//
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, { headers: this._headers })
+      .then(this._checkResponse)
+      .catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`    after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.